modules/silicon/scripts/usage/ai-notebook-desktop-script.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+
+# Copyright 2021 Google LLC
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+#   ___  _____   _   _       _       _                 _
+#  / _ \|_   _| | \ | |     | |     | |               | |
+# / /_\ \ | |   |  \| | ___ | |_ ___| |__   ___   ___ | | __
+# |  _  | | |   | . ` |/ _ \| __/ _ \ '_ \ / _ \ / _ \| |/ /
+# | | | |_| |_  | |\  | (_) | ||  __/ |_) | (_) | (_) |   <
+# \_| |_/\___/  \_| \_/\___/ \__\___|_.__/ \___/ \___/|_|\_\
+
+
+
+export INSTANCE_NAME="AI_NOTEBOOK_INSTANCE_NAME"
```

Comment:
couldn't those be inferred from the environment instead?

Reply:
Yes, we can.  Can you update it to this?

```shell
export INSTANCE_NAME=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/name)
export PROJECT=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/project/project-id)
export ZONE=$(curl -s -H "Metadata-Flavor: Google" http://metadata.google.internal/computeMetadata/v1/instance/zone | cut -d'/' -f 4)
```

Reply:
I think this script can also run on the user workstation (per the `README.md` in the same dir) so I don't think we could query the metadata server if they're not using cloud shell.

Maybe we could simply inline the gcloud cli invocation in the README? or deduce them from the environment/gcloud configuration/parameters from the script?

Reply:
Correct, ignore this.